control_plane/protobuf_utils.h

```diff
@@ -19,7 +19,8 @@ namespace P4Tools::Flay::Protobuf {
 /// (boost cpp_int).
 inline big_int stringToBigInt(const std::string &valueString) {
     big_int value;
-    boost::multiprecision::import_bits(value, valueString.begin(), valueString.end());
+    std::vector<uint8_t> Uint8Vec(valueString.begin(), valueString.end());
```

Comment:
What happens when we use std::string_view as parameter instead of `const std::string&`? Does it have a similar effect?

Reply:
I think it is the problem of character type, which is signed. But a u_char string_view does help.

```cpp
#include <boost/multiprecision/cpp_int.hpp>
#include <iostream>
#include <vector>

namespace mp = boost::multiprecision;

void import_and_print(const std::string &s) {
    mp::cpp_int value;
    mp::import_bits(value, s.begin(), s.end());
    std::cout << "std::string: " << value << std::endl;
}

void import_and_print(const std::string_view s) {
    mp::cpp_int value;
    mp::import_bits(value, s.begin(), s.end());
    std::cout << "std::string_view: " << value << std::endl;
}

void import_and_print(const std::basic_string_view<u_char> s) {
    mp::cpp_int value;
    mp::import_bits(value, s.begin(), s.end());
    std::cout << "std::basic_string_view<u_char>: " << value << std::endl;
}

void import_and_print(const std::vector<uint8_t> &uint8_vec) {
    mp::cpp_int value;
    mp::import_bits(value, uint8_vec.begin(), uint8_vec.end());
    std::cout << "std::vector<uint8_t>:" << value << std::endl;
}

int main() {
    std::string s;
    s.resize(2);
    s[0] = 'y';
    s[1] = '\032';
    
    import_and_print(s); // std::string: 15514

    std::string_view sv(s);
    import_and_print(sv); // std::string_view: 15514

    std::basic_string_view<u_char> u_char_sv(reinterpret_cast<const u_char *>(s.data()), s.size());
    import_and_print(u_char_sv); // std::basic_string_view<u_char>: 31002

    std::vector<uint8_t> uint8_vec;
    for (auto c : s) {
        uint8_vec.push_back(c);
    }
    import_and_print(uint8_vec); // std::vector<uint8_t>:31002

    return 0;
}
```

Reply:
Instead of transform the string into a vector, we may use `mp::import_bits(value, (const u_char *) s.data(), (const u_char *) s.data() + s.size());` similar to what basic_string_view<u_char> does.

Reply:
The last approach seems like the best one. There is little we can do about the protobuf data I believe. Which we could just get a byte array there.

Reply:
Yeah, I pushed the code using the last approach.